cygnus-twitter/src/main/java/com/telefonica/iot/cygnus/sinks/TwitterHDFSSink.java

```diff
@@ -0,0 +1,380 @@
+                + ")");
+        krb5ConfFile = context.getString("krb5_auth.krb5_conf_file", "");
+        LOGGER.debug("[" + this.getName() + "] Reading configuration (krb5_conf_file=" + krb5ConfFile + ")");
+
+        String backendImplStr = context.getString("backend_impl", "rest");
+
+        try {
+            backendImpl = BackendImpl.valueOf(backendImplStr.toUpperCase());
+            LOGGER.debug("[" + this.getName() + "] Reading configuration (backend_impl="
+                        + backendImplStr + ")");
+        } catch (Exception e) {
+            invalidConfiguration = true;
+            LOGGER.debug("[" + this.getName() + "] Invalid configuration (backend_impl="
+                + backendImplStr + ") -- Must be 'rest' or 'binary'");
+        }  // try catch
+        
+        super.configure(context);
+
+    } // configure
+
+    @Override
+    public void start() {
+        try {
+            // create the persistence backend
+            if (backendImpl == BackendImpl.BINARY) {
+                persistenceBackend = new HDFSBackendImplBinary(host, port, username, password, oauth2Token,
+                        "", "", "", enableKrb5, krb5User, krb5Password, krb5LoginConfFile,
+                        krb5ConfFile, false);
+            } else if (backendImpl == BackendImpl.REST) {
+                persistenceBackend = new HDFSBackendImplREST(host, port, username, password, oauth2Token,
+                        "", "", "", enableKrb5, krb5User, krb5Password, krb5LoginConfFile,
+                        krb5ConfFile, false);
+            } else {
+                LOGGER.fatal("The configured backend implementation does not exist, Cygnus will exit. Details="
+                        + backendImpl.toString());
+                System.exit(-1);
+            } // if else
+
+            LOGGER.debug("[" + this.getName() + "] HDFS persistence backend created");
+        } catch (Exception e) {
+            LOGGER.error("Error while creating the HDFS persistence backend. Details="
+                    + e.getMessage());
+        } // try catch
+
+        super.start();
+    } // start
+
+    @Override
+    void persistBatch(TwitterBatch batch) throws Exception {
+        if (batch == null) {
+            LOGGER.debug("[" + this.getName() + "] Null batch, nothing to do");
+            return;
+        } // if
+
+        // get an aggregator for this destination and initialize it
+        HDFSAggregator aggregator = new JSONRowAggregator();
+
+        ArrayList<TwitterEvent> subBatch = batch.getBatch();
+        aggregator.initialize(hdfs_folder, hdfs_file);
+        for (TwitterEvent cygnusEvent : subBatch) {
+            aggregator.aggregate(cygnusEvent);
+        } // for
+
+        // persist the aggregation
+        persistAggregation(aggregator);
+
+
+    } // persistBatch
+
+    /**
+     * Class for aggregating aggregation.
+     */
+    private abstract class HDFSAggregator {
```

Comment:
Since the only class inheriting this `HDFSAggregator` (which is an abstract class) is `JsonRowAggregator`, I think you only need a non-abstract `HDFSAggregator` class implementing the logic of `JsonRowAggregator`.